.github/workflows/labels.yml

```diff
@@ -27,34 +32,86 @@ jobs:
     steps:
+      - uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7.0.1
+        id: prepare
+        with:
+          script: |
+            if (context.payload.workflow_run) {
+              // PRs from forks don't have any PRs associated by default.
+              // Thus, we request the PR number with an API call *to* the fork's repo.
+              const { head_repository, head_sha } = context.payload.workflow_run
+              const { number } = (await github.rest.repos.listPullRequestsAssociatedWithCommit({
+                owner: head_repository.owner.login,
+                repo: head_repository.name,
+                commit_sha: head_sha
+              })).data[0]
+              core.setOutput('pr-number', number)
+            } else {
+              core.setOutput('pr-number', context.payload.pull_request.number)
+            }
+
+            const run_id = (await github.rest.actions.listWorkflowRuns({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              workflow_id: 'pr.yml',
+              event: 'pull_request_target',
+              head_sha: context.payload.pull_request?.head.sha ?? context.payload.workflow_run.head_sha
+            })).data.workflow_runs[0]?.id
+            core.setOutput('run-id', run_id)
+
       - name: Download the comparison results
+        if: steps.prepare.outputs.run-id
         uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4.1.8
         with:
           pattern: comparison
           path: comparison
           merge-multiple: true
+          run-id: ${{ steps.prepare.outputs.run-id }}
 
       - name: Labels from eval
-        if: ${{ github.event_name != 'pull_request' }}
+        if: steps.prepare.outputs.run-id && github.event_name != 'pull_request'
         uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7.0.1
+        env:
+          PULL_NUMBER: ${{ steps.prepare.outputs.pr-number }}
         with:
           script: |
             const { readFile } = require('node:fs/promises')
+            const pull_number = process.env.PULL_NUMBER
 
             const pr = {
               owner: context.repo.owner,
               repo: context.repo.repo,
-              issue_number: context.payload.pull_request.number
+              issue_number: pull_number
             }
 
             // Get all currently set labels that we manage
             const before =
               (await github.paginate(github.rest.issues.listLabelsOnIssue, pr))
               .map(({ name }) => name)
-              .filter(name => name.startsWith('10.rebuild') || name == '11.by: package-maintainer')
+              .filter(name =>
+                name.startsWith('10.rebuild') ||
+                name == '11.by: package-maintainer' ||
+                name.startsWith('12.approvals:') ||
+                name.startsWith('12.approved-by: package-maintainer')
```

Comment:
Nit: `11.by` and `12.approved-by` should be consistent in either using `==` or `startsWith`, since both know the entire label name.